setup/package.py

```diff
@@ -58,19 +58,25 @@ def run_cmd(cmd, chdir=None, timeout=None):
     return subprocess.run(cmd, cwd=chdir, timeout=timeout)
 
 
+def _database_status(addr, port, dbname, *, state=None):
+    url = f'{addr}:{port}/web/database/status'
+    res = requests.post(url, data={'master_pwd': 'admin', 'name': dbname})
+    if res.status_code != 200:
+        exc = OdooTestError(f"{url} {res.status_code}")
+        exc.add_note(res.text)
+        raise exc
+    modules = res.json()
+    if state is None:
+        return modules
+    return [mod for mod in modules if mod['state'] == state]
```

Comment:
Do the filter in `_rpc_count_modules`, this removes the need for the state parameter and you can just have 1 remote call.

Reply:
There is only one call to that function (in `test_odoo`) and it already sleeps 5 before calling it. There is also no point of having sleep 1 between rpc calls.
I think you can remove all sleep calls in that function - unless you can explain why they are needed here. Just have one call that lists all "installed" and "to install" modules at once and filter the result.

Reply:
I'm calling @d-fence , cause it is his script, not mine :smile: